ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java

```diff
@@ -0,0 +1,132 @@
+                .collect(Collectors.toList()))
+            .build();
+
+        addUploadDocumentDirection(directionOrderData, directionOrder);
+        addEyeWitnessDirection(directionOrderData, directionOrder);
+
+        asStream(directionOrderData.getOtherDirections())
+            .map(CCDCollectionElement::getValue)
+            .forEach(ccdOrderDirection -> directionOrder.addDirection(Direction.builder()
+                .directionType(DirectionType.valueOf(ccdOrderDirection.getExtraOrderDirection().name()))
+                .directionComment(ccdOrderDirection.getDirectionComment())
+                .directionParty(DirectionParty.valueOf(ccdOrderDirection.getForParty().name()))
+                .directionActionedBy(ccdOrderDirection.getSendBy())
+                .directionHeaderType(getDirectionHeaderType(ccdOrderDirection))
+                .expertReports(asStream(ccdOrderDirection.getExpertReports())
+                    .filter(Objects::nonNull)
+                    .map(CCDCollectionElement::getValue)
+                    .collect(Collectors.toList()))
+                .extraDocuments(asStream(ccdOrderDirection.getExtraDocUploadList())
+                    .map(CCDCollectionElement::getValue)
+                    .collect(Collectors.toList()))
+                .build()));
+
+        return directionOrder;
+    }
+
+    private DirectionHeaderType getDirectionHeaderType(CCDOrderDirection ccdOrderDirection) {
+        return ccdOrderDirection.getOtherDirectionHeaders() == null
+            ? null
+            : DirectionHeaderType.valueOf(ccdOrderDirection.getOtherDirectionHeaders().name());
+    }
+
+    private void addEyeWitnessDirection(CCDOrderGenerationData directionOrderData, DirectionOrder directionOrder) {
+        directionOrder.addDirection(mapDirectionData(directionOrderData, CCDOrderDirectionType.EYEWITNESS));
+    }
+
+    private void addUploadDocumentDirection(CCDOrderGenerationData directionOrderData, DirectionOrder directionOrder) {
+        directionOrder.addDirection(mapDirectionData(directionOrderData, CCDOrderDirectionType.DOCUMENTS));
+    }
+
+    private Direction mapDirectionData(CCDOrderGenerationData directionOrderData, CCDOrderDirectionType directionType) {
+        Direction.DirectionBuilder builder = Direction.builder();
+        mapDirectionForType(directionOrderData, builder, directionType);
+        return builder.build();
+    }
+
+    private void mapDirectionForType(
+        CCDOrderGenerationData directionOrderData,
+        Direction.DirectionBuilder builder,
+        CCDOrderDirectionType directionType
+    ) {
+        switch (directionType) {
+            case DOCUMENTS:
+                addDirectionParty(builder, directionOrderData.getDocUploadForParty());
+
+                builder.directionType(DirectionType.valueOf(directionType.name()))
+                    .directionActionedBy(directionOrderData.getDocUploadDeadline());
+                return;
```

Comment:
I think `break` would be more appropriate in this switch.

Reply:
changed